test/shared/src/main/scala/zio/test/DefaultTestReporter.scala

```diff
@@ -19,75 +19,89 @@ package zio.test
 import zio.test.RenderedResult.CaseType._
 import zio.test.RenderedResult.Status._
 import zio.test.RenderedResult.{ CaseType, Status }
-import zio.{ Cause, ZIO }
+import zio.{ Cause, UIO, URIO, ZIO }
 import scala.{ Console => SConsole }
 import zio.duration.Duration
 
 object DefaultTestReporter {
 
-  def render[E, S](executedSpec: ExecutedSpec[String, E, S]): Seq[RenderedResult] = {
-    def loop(executedSpec: ExecutedSpec[String, E, S], depth: Int): Seq[RenderedResult] =
+  def render[E, S](executedSpec: ExecutedSpec[String, E, S]): UIO[Seq[RenderedResult]] = {
+    def loop(executedSpec: ExecutedSpec[String, E, S], depth: Int): UIO[Seq[RenderedResult]] =
       executedSpec.caseValue match {
         case Spec.SuiteCase(label, executedSpecs, _) =>
-          val hasFailures = executedSpecs.exists(_.exists {
-            case Spec.TestCase(_, test) => test.isLeft; case _ => false
-          })
-          val status        = if (hasFailures) Failed else Passed
-          val renderedLabel = if (hasFailures) renderFailureLabel(label, depth) else renderSuccessLabel(label, depth)
-          rendered(Suite, label, status, depth, renderedLabel) +: executedSpecs.flatMap(loop(_, depth + tabSize))
+          for {
+            specs <- executedSpecs
+            failures <- UIO.foreach(specs)(_.exists {
+                         case Spec.TestCase(_, test) => test.map(_.isLeft);
+                         case _                      => UIO.succeed(false)
+                       })
+            hasFailures   = failures.exists(identity)
+            status        = if (hasFailures) Failed else Passed
+            renderedLabel = if (hasFailures) renderFailureLabel(label, depth) else renderSuccessLabel(label, depth)
+            rest          <- UIO.foreach(specs)(loop(_, depth + tabSize)).map(_.flatten)
```

Comment:
Fixed to sequential here

Reply:
Similar thought to the above that since we've already run the tests we don't need parallelism in rendering the results.